br/pkg/task/operator/cmd.go

```diff
@@ -35,56 +38,129 @@ func dialPD(ctx context.Context, cfg *task.Config) (*pdutil.PdController, error)
+	cleanUpWithErr(func(ctx context.Context) error { f(ctx); return nil })
+}
+
+func cleanUpWithErr(f func(ctx context.Context) error) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
-	f(ctx)
+	return f(ctx)
 }
 
-// PauseGCAndScheduler blocks the current goroutine and pause the GC safepoint and remove the scheduler by the config.
+type AdaptEnvForSnapshotBackupContext struct {
+	context.Context
+
+	pdMgr *pdutil.PdController
+	kvMgr *utils.StoreManager
+	cfg   PauseGcConfig
+
+	rdGrp  sync.WaitGroup
+	runGrp *errgroup.Group
+}
+
+func (ctx *AdaptEnvForSnapshotBackupContext) ReadyL(name string, notes ...zap.Field) {
+	logutil.CL(ctx).Info("Stage ready.", append(notes, zap.String("component", name))...)
+	ctx.rdGrp.Done()
+}
+
+func hintAllReady() {
+	// Hacking: some version of operators using the follow two logs to check whether we are ready...
+	log.Info("Schedulers are paused.")
+	log.Info("GC is paused.")
+	log.Info("All ready.")
+}
+
+// AdaptEnvForSnapshotBackup blocks the current goroutine and pause the GC safepoint and remove the scheduler by the config.
 // This function will block until the context being canceled.
-func PauseGCAndScheduler(ctx context.Context, cfg *PauseGcConfig) error {
+func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig) error {
 	mgr, err := dialPD(ctx, &cfg.Config)
 	if err != nil {
-		return err
+		return errors.Annotate(err, "failed to dial PD")
 	}
-
+	var tconf *tls.Config
+	if cfg.TLS.IsEnabled() {
+		tconf, err = cfg.TLS.ToTLSConfig()
+		if err != nil {
+			return errors.Annotate(err, "invalid tls config")
+		}
+	}
+	kvMgr := utils.NewStoreManager(mgr.GetPDClient(), keepalive.ClientParameters{
+		Time:    time.Duration(cfg.Config.GRPCKeepaliveTime) * time.Second,
+		Timeout: time.Duration(cfg.Config.GRPCKeepaliveTimeout) * time.Second,
+	}, tconf)
 	eg, ectx := errgroup.WithContext(ctx)
+	cx := &AdaptEnvForSnapshotBackupContext{
+		Context: ectx,
+		pdMgr:   mgr,
+		kvMgr:   kvMgr,
+		cfg:     *cfg,
+		rdGrp:   sync.WaitGroup{},
+		runGrp:  eg,
+	}
+	cx.rdGrp.Add(3)
 
-	eg.Go(func() error { return pauseGCKeeper(ectx, cfg, mgr) })
-	eg.Go(func() error { return pauseSchedulerKeeper(ectx, mgr) })
+	eg.Go(func() error { return pauseGCKeeper(cx) })
+	eg.Go(func() error { return pauseSchedulerKeeper(cx) })
+	eg.Go(func() error { return goPauseImporting(cx) })
+	go func() {
+		cx.rdGrp.Wait()
+		hintAllReady()
+	}()
 
 	return eg.Wait()
 }
 
-func pauseGCKeeper(ctx context.Context, cfg *PauseGcConfig, ctl *pdutil.PdController) error {
+func goPauseImporting(ctx *AdaptEnvForSnapshotBackupContext) error {
+	denyLightning := utils.NewDenyImporting("prepare_for_snapshot_backup", ctx.kvMgr)
+	if _, err := denyLightning.DenyAllStores(ctx, ctx.cfg.TTL); err != nil {
+		return errors.Trace(err)
+	}
+	ctx.ReadyL("pause_lightning")
+	ctx.runGrp.Go(func() error {
+		err := denyLightning.Keeper(ctx, ctx.cfg.TTL)
+		if errors.Cause(err) != context.Canceled {
```

Comment:
what's the value to record a log entry here?

Reply:
So we can know what happened to TiKV for troubleshooting.